src/stp/new_plays/ReflectKick.cpp

```diff
@@ -0,0 +1,132 @@
+                                                                                 std::make_unique<role::Formation>(role::Formation("offender_2")),
+                                                                                 std::make_unique<role::Formation>(role::Formation("midfielder_1")),
+                                                                                 std::make_unique<role::Formation>(role::Formation("midfielder_2")),
+                                                                                 std::make_unique<role::Formation>(role::Formation("midfielder_3")),
+                                                                                 std::make_unique<role::Defender>(role::Defender("defender_1")),
+                                                                                 std::make_unique<role::Defender>(role::Defender("defender_2")),
+                                                                                 std::make_unique<role::Defender>(role::Defender("defender_3"))};
+}
+
+uint8_t ReflectKick::score(world_new::World *world) noexcept { return 500; }
+
+Dealer::FlagMap ReflectKick::decideRoleFlags() const noexcept {
+    Dealer::FlagMap flagMap;
+    Dealer::DealerFlag keeperFlag(DealerFlagTitle::KEEPER, DealerFlagPriority::KEEPER);
+    Dealer::DealerFlag closeToBallFlag(DealerFlagTitle::CLOSE_TO_BALL, DealerFlagPriority::HIGH_PRIORITY);
+    Dealer::DealerFlag closeToTheirGoalFlag(DealerFlagTitle::CLOSE_TO_THEIR_GOAL, DealerFlagPriority::MEDIUM_PRIORITY);
+    Dealer::DealerFlag closeToOurGoalFlag(DealerFlagTitle::CLOSE_TO_OUR_GOAL, DealerFlagPriority::MEDIUM_PRIORITY);
+    Dealer::DealerFlag not_important(DealerFlagTitle::ROBOT_TYPE_50W, DealerFlagPriority::LOW_PRIORITY);
+
+    flagMap.insert({"keeper", {keeperFlag}});
+    flagMap.insert({"reflecter", {closeToTheirGoalFlag}});
+    flagMap.insert({"passer", {closeToBallFlag}});
+    flagMap.insert({"offender_1", {not_important}});
+    flagMap.insert({"offender_2", {not_important}});
+    flagMap.insert({"midfielder_1", {not_important}});
+    flagMap.insert({"midfielder_2", {not_important}});
+    flagMap.insert({"midfielder_3", {not_important}});
+    flagMap.insert({"defender_1", {closeToOurGoalFlag}});
+    flagMap.insert({"defender_2", {closeToOurGoalFlag}});
+    flagMap.insert({"defender_3", {closeToOurGoalFlag}});
+
+    return flagMap;
+}
+
+void ReflectKick::calculateInfoForRoles() noexcept {
+    // TODO: Change this
+    auto passPosition = Vector2();
+
+    auto ball = world->getWorld()->getBall().value();
+    std::optional<Vector2> intersection;
+
+    if ((ball->getPos() - passPosition).length() <= 2.0 && ball->getVelocity().length() > 0.1) {
+        intersection = Line(ball->getPos(), ball->getPos() + ball->getVelocity()).intersects(
+                Line(passPosition, field.getTheirGoalCenter()));
+    }
```

Comment:
This will be problematic if the ball ever rolls towards our goal, there will still be an intersection.

Reply:
Lines are infinite